### PR TITLE
fix: deploy only once if two projects share the same job

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: src/commands/run/jobs-builder/deploy-job-builder.ts [hunk 1]
```diff
@@ -36,6 +36,31 @@ export class DeployJobBuilder {
...
+  private mergeDoubledProjects(projects: JobDescriptor[]) {
+    const filteredProjects = new Map<string, JobDescriptor>();
+
+    projects.forEach(project => {
+      const currentProject = filteredProjects.get(project.opts.name);
+
+      if (currentProject) {
+        Object.keys(project.opts.parameters).forEach(key => {
+          if (
+            project.opts.parameters[key] !== currentProject.opts.parameters[key] &&
```
Comment (Loczi94) [inline src/commands/run/jobs-builder/deploy-job-builder.ts:52]: How about creating two variables to store `project.ops.paremeters` and `currentProject.opt.parameters`? Now it's a little bit unclear. What do you think?

File: src/commands/run/jobs-builder/deploy-job-builder.ts [hunk 2]
```diff
@@ -36,6 +36,31 @@ export class DeployJobBuilder {
           parameters: this.mapProjectParams(project, params),
         },
       }));
+
+    return this.mergeDoubledProjects(mappedProjects);
+  }
+
+  private mergeDoubledProjects(projects: JobDescriptor[]) {
```
[the remark this answers is not in the document]
Comment (dmnsobczak) [reply src/commands/run/jobs-builder/deploy-job-builder.ts:43]: You right! Added!

File: src/commands/run/jobs-builder/deploy-job-builder.ts (hunk 1 above)
Comment (dmnsobczak) [reply src/commands/run/jobs-builder/deploy-job-builder.ts:52]: Well, I disagree. We're using not only `project.ops.parameters` but also `project.displayName`. Creating variables for parameters only will still require to use the original object and I'm not a fan of that mix. I can create 4 new variables but that's overkill I guess. Unless you have other, specific proposition? Or maybe renaming the existing variables would help?

Comment (Loczi94) [reply src/commands/run/jobs-builder/deploy-job-builder.ts:52]: Okay, I agree.

**Comment by dmnsobczak** [comment]
You chose the approach which I wanted to ignore, but we discussed this offline (via Slack 🙈) and I'm agree with you.

**Pull request merged**